### PR TITLE
[website] Stop publishing pre 3.5 API docs, clean up deploy flow

### PR DESCRIPTION
**Pull request by mwachs5**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

  - performance improvement   

#### API Impact

No impact on APIs other than making pre-3.5 APIs harder to find.

#### Backend Code Generation Impact

No change

#### Desired Merge Strategy

  - Squash: The PR will be squashed and merged (choose this if you have no preference.

#### Release Notes

website: Stop publishing pre-3.5 API Docs

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.4.x`, [small] API extension: `3.5.x`, API modification or big change: `3.6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by mwachs5** [comment]
Currently tries to address #2947 by just ignoring contributors that could not be fetched.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -252,3 +206,23 @@ jobs:
           PGP_SECRET: ${{ secrets.PGP_SECRET }}
           SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
           SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
+
+
+  deploy_website:
+    name: Deploy Website
+    runs-on: ubuntu-latest
+    needs: [all_tests_passed]
+    if: (github.event_name == 'push') && (github.ref_type == 'branch') && ((github.ref_name == 'website-test') || (github.ref_name == 'master')
```
Comment (mwachs5) [inline .github/workflows/test.yml:215]:
```suggestion
    if: (github.event_name == 'push') && (github.ref_type == 'branch') && ((github.ref_name == 'website-test') || (github.ref_name == 'master'))
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -169,33 +129,27 @@ jobs:
...
         run: make -C website
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - name: Deploy Website to GitHub Pages (From Master Branch)
-        if: (github.event_name == 'push') && (github.ref_type == 'branch') && (github.ref_name == 'master')
-        uses: JamesIves/github-pages-deploy-action@3.7.1
+      - name: Tar built website
+        run: tar zcf website.tar.gz website/docs/target/site
+      - name: Share Built Website
+        uses: actions/upload-artifact@v2
```
Comment (jackkoenig) [inline .github/workflows/test.yml:141]:
```suggestion
        uses: actions/upload-artifact@v3
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -252,3 +206,23 @@ jobs:
...
+
+
+  deploy_website:
+    name: Deploy Website
+    runs-on: ubuntu-latest
+    needs: [all_tests_passed]
+    if: (github.event_name == 'push') && (github.ref_type == 'branch') && ((github.ref_name == 'website-test') || (github.ref_name == 'master'))
+    steps:
+      - name: Download built website
+        uses: actions/download-artifact@v2
```
Comment (jackkoenig) [inline .github/workflows/test.yml:218]:
```suggestion
        uses: actions/download-artifact@v3
```

File: .github/workflows/test.yml (hunk 2 above)
Comment (mwachs5) [reply .github/workflows/test.yml:141]: what does this mean? Is it a new artifact or just a new version of the action?

**Review by seldridge — APPROVED** [review]
Nice.

Is there any impact on the availability of the pre 3.5 docs?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -110,48 +110,8 @@ jobs:
...
-      - name: Share API Docs
-        uses: actions/upload-artifact@v2
-        with:
-          name: ${{ matrix.project }}-api-docs
-          path: website/build/api/${{ matrix.project }}.tar.gz
-
   website:
-    name: Build & Deploy Website
-    needs: [website-api-docs]
+    name: Build Website
```
Comment (jackkoenig) [inline .github/workflows/test.yml:114]:
```suggestion
    name: Build Website (and API Docs)
```

Please call out that building the website also builds the API docs--I will forget and then add a new test in the future if this isn't mentioned 🙂

**Review by jackkoenig — APPROVED** [review]
LGTM with 1 ask

File: .github/workflows/test.yml (hunk 4 above)
Comment (mwachs5) [reply .github/workflows/test.yml:114]: its not the API docs, it's the Mdocs, do you agree?

Comment (mwachs5) [inline .github/workflows/test.yml:114]:
```suggestion
    name: Build Website and Mdoc-Based Documentation
```

Comment (mwachs5) [inline .github/workflows/test.yml:114]:
```suggestion
    name: Build Mdoc & Website
```

**Pull request merged**